packages/stems/src/components/HarmonyButton/HarmonyButton.tsx

```diff
@@ -0,0 +1,105 @@
+import { CSSProperties, forwardRef } from 'react'
+
+import cn from 'classnames'
+
+import { useMediaQueryListener } from 'hooks/useMediaQueryListener'
+import { CSSCustomProperties } from 'styles/types'
+import { toCSSVariableName } from 'utils/styles'
+
+import styles from './HarmonyButton.module.css'
+import {
+  HarmonyButtonProps,
+  HarmonyButtonType,
+  HarmonyButtonSize
+} from './types'
+
+const SIZE_STYLE_MAP: { [k in HarmonyButtonSize]: string } = {
+  [HarmonyButtonSize.SMALL]: styles.small,
+  [HarmonyButtonSize.DEFAULT]: styles.default,
+  [HarmonyButtonSize.LARGE]: styles.large
+}
+
+const TYPE_STYLE_MAP: { [k in HarmonyButtonType]: string } = {
+  [HarmonyButtonType.PRIMARY]: styles.primary,
+  [HarmonyButtonType.SECONDARY]: styles.secondary,
+  [HarmonyButtonType.TERTIARY]: styles.tertiary,
+  [HarmonyButtonType.DESTRUCTIVE]: styles.destructive,
+  [HarmonyButtonType.GHOST]: styles.ghost
+}
+
+/**
+ * A common Button component. Includes a few variants and options to
+ * include and position icons.
+ */
+export const HarmonyButton = forwardRef<HTMLButtonElement, HarmonyButtonProps>(
+  function Button(
+    {
+      color,
+      text,
+      variant = HarmonyButtonType.PRIMARY,
+      size = HarmonyButtonSize.DEFAULT,
+      leftIcon: LeftIconComponent,
```

Comment:
headsup mobile does `iconLeft`. im fine with either, but prob want consistency. if we go with leftIcon, can we make a note in mobile code on the prop definitions that we'd like to rename?

Reply:
Oh I'd rather this be consistent. Can switch it to iconLeft. I think it actually works better because then `iconLeft` and `iconRight` sort next to each other lexographically.